django_emarsys/apps.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+
+from django.apps import AppConfig
+from django.core.checks import register
+
+from django_emarsys.config import validate_settings
+
+
+class DjangoEmarsysConfig(AppConfig):
+    name = 'django_emarsys'
+    verbose_name = "Django Emarsys"
+
+    def ready(self):
+        @register()
+        def check_settings(app_configs):
+            messages = validate_settings()
+            for message in messages:
+                # apparently Django's check framework can't handle
+                # unicode correctly, so replace all non-ascii characters
+                # with '?'
+                message.msg = re.sub(r'[^\x00-\x7F]+', '?', message.msg)
```

Comment:
Ah, nice. Had forgotten about `'replace'`.

Event names and parameter names can and do contain non-ASCII characters ("Email bestätigen", etc.).

It has nothing to do with stdout:
```
Performing system checks...

Unhandled exception in thread started by <function wrapper at 0x7fadb1a35500>
Traceback (most recent call last):
  File "/home/thi/.virtualenvs/machtfit-oscar/local/lib/python2.7/site-packages/django/utils/autoreload.py", line 222, in wrapper
    fn(*args, **kwargs)
  File "/home/thi/.virtualenvs/machtfit-oscar/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 105, in inner_run
    self.validate(display_num_errors=True)
  File "/home/thi/.virtualenvs/machtfit-oscar/local/lib/python2.7/site-packages/django/core/management/base.py", line 362, in validate
    return self.check(app_configs=app_configs, display_num_errors=display_num_errors)
  File "/home/thi/.virtualenvs/machtfit-oscar/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in check
    formatted = "\n".join(sorted(formatted))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 87: ordinal not in range(128)
```
Django tries to output messages in `django.core.management.base`, also using `force_str`, so each item in `formatted` will be a utf-8 encoded `str`. Even if it didn't, tho, `CheckMessage` has the decorator `python_2_unicode_compatible`, which overwrites its `__str__` to return a utf-8 encoded `str` already.

But `base` also has `unicode_literal` imported, so the `"\n"` in `"\n".join(sorted(formatted))` is `unicode` and adding to it will attempt to decode every `str` in the iterable `.join()` encounters. Unfortunately it uses the default encoding for this, which is `ascii`.

Googling reveals quite a few strange things about that. In short: there's `sys.setdefaultencoding()`, but Python's own `site.py` deletes it early on.
A work-around is this:
```
import sys
reload(sys)
sys.setdefaultencoding('utf-8')
```
And then indeed it works. The problem is that I'm not at all sure that this won't break something else, which might rely on the other default encoding, so I think it's enough to just hide the non-ASCII chars from our check messages.